### ISSUE TITLE
Day component

### ISSUE THREAD

**Issue by atherdon** [GroceriStar/react-only-intern-23#26]
I think it should be rewritten. Tell me how do you think it can be done by your opinion

**Comment by sanchit94** [comment]
Alright @atherdon  so by this you mean making it more modular, right? What we can do is make a `method` for rendering each vertical column. This method will take an object as argument which contains all the data that is to be displayed, and then by parsing the time duration, we can use it to figure out the height of each clickable link, and also where it can be placed in the column. At other places where there is no clickable link, the column rendering method will automatically render empty list items. This will make WorkSched very modular in a way that you can run the column rendering method with the data and the method will figure everything out.

**Comment by sanchit94** [comment]
@atherdon Okay so it took a while to get this done. I wrote the code for this three times and this is the attempt in which I think I was able to achieve what I planned for. Do review the code and see if there is some issue with it!

**Comment by atherdon** [comment]
sure, i'll


On 27-Jun-2018 at 09:31 PM, Sanchit Sharma wrote:


@atherdon <https://github.com/atherdon> Okay so it took a while to get this
done. I wrote the code for this three times and this is the attempt in
which I think I was able to achieve what I planned for. Do review the code
and see if there is some issue with it!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/GroceriStar/react-only-intern-23/issues/26#issuecomment-400835627>,
or mute the thread
